homework/index.js

```diff
@@ -30,17 +15,135 @@
+  }
+
+  function renderRepository(tableDiv, repository) {
+    tableDiv.innerHTML = '';
+
+    const table = createAndAppend('table', tableDiv);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository:', 'td-header', ' ');
+    createAndAppend('a', firstRow.lastChild, {
+      text: repository.name,
+      href: repository.html_url,
+      target: '_blank',
+    });
+    if (repository.description) {
+      addRow(tbody, 'Description:', 'description-td td-header', repository.description);
+    }
+    addRow(tbody, 'Forks:', 'td-header', repository.forks_count);
+    addRow(tbody, 'Updated:', 'td-header', new Date(repository.updated_at).toLocaleString());
+  }
+
+  async function renderContributions(url, contributesDiv) {
+    try {
+      const contributors = await fetchJSON(url);
+      contributesDiv.innerHTML = '';
+      createAndAppend('h1', contributesDiv, {
+        class: 'contributes-header',
+        text: 'Contributions',
+      });
+      const ul = createAndAppend('ul', contributesDiv);
+
+      contributors.forEach(contributor => {
+        const listItem = createAndAppend('li', ul, { class: 'list flex-div' });
+        const hyperlink = createAndAppend('a', listItem, {
+          href: contributor.html_url,
+          target: '_blank',
+        });
+        createAndAppend('img', hyperlink, {
+          src: contributor.avatar_url,
+          alt: `${contributor.login} photo`,
+        });
+
+        const contributorInfoDiv = createAndAppend('div', hyperlink, {
+          class: 'contributor-info flex-div',
+        });
+        createAndAppend('p', contributorInfoDiv, {
+          class: 'contributorsName',
+          text: contributor.login,
+        });
+        createAndAppend('p', contributorInfoDiv, {
+          text: contributor.contributions,
+          class: 'badge',
+        });
+      });
+    } catch (error) {
+      renderError(error);
+    }
+  }
+
+  function createOptionElements(repositories, select) {
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, { text: repository.name, value: index });
+    });
+  }
+  function renderEventListener(repositories, select, tableDiv, contributesDiv) {
+    select.addEventListener('change', () => {
+      loader(contributesDiv);
+      renderRepository(tableDiv, repositories[select.value]);
+      renderContributions(repositories[select.value].contributors_url, contributesDiv);
+    });
+  }
+
+  async function renderDefaultPage(url, select, tableDiv, contributesDiv) {
+    let repositories;
+    try {
+      const data = await fetchJSON(url);
+      repositories = data.sort((a, b) => a.name.localeCompare(b.name));
+      createOptionElements(repositories, select);
+      renderRepository(tableDiv, repositories[select.value]);
+      renderContributions(repositories[select.value].contributors_url, contributesDiv);
+      renderEventListener(repositories, select, tableDiv, contributesDiv);
```

Comment:
The verb `render` in IT is generally understood as visualizing data on a medium. In this case, where you are not visualizing anything, a better name would be `addEventListener`, were it not that this name is too easily confused with `element.addEventListener`.

I think I would create a function, say `createHeader`, in which I would do all the header related stuff, including adding the event listener to the `select` element. Then I would call `createHeader()` from `main()`.

Reply:
Thanks a lot for taking the time and effort in reviewing my homework! And I am very happy that you accept my PR as-is!